tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -116,6 +116,17 @@ static cl::opt<bool> SPIRVGenKernelArgNameMD(
     cl::desc("Enable generating OpenCL kernel argument name "
              "metadata"));
 
+// TODO: rename this option
+static cl::opt<SPIRV::BIsRepresentation> BIsRepresentation(
+    "spirv-ocl-builtins-version",
+    cl::desc("Specify version of OCL builtins to translate "),
```

Comment:
Since the option is now more generic than just OCL builtins version, we should probably rename it. @svenvh, @Naghasan, any suggestions?

Reply:
How about (probably) the shortest and simplest: `--builtins`?

Reply:
+1 this keeps it neutral so probably the best. Probably better to have `spirv-builtins` for uniformity.

Reply:
What about `--builtins-representation`? Or `--bi(s)-representation`?

To me, personally, just `builtins` looks a bit unclear, what is builtins, which builtins?

Reply:
+1 for keeping uniformity so `spirv-builtins` would be better.  Although that could be confusing because it can also be interpreted as "SPIR-V builtins"?

I don't think adding the word "representation" answers the question "what/which builtins".

Another idea: `spirv-builtins-target` or `spirv-target-builtins`?

Reply:
> I don't think adding the word "representation" answers the question "what/which builtins".

Together with proper description it should be clear that this option is about how to represent "SPIRV builtins".

> Although that could be confusing because it can also be interpreted as "SPIR-V builtins"?

I agree here. What about even more generic `spirv-ops-target`, `spirv-ops-representation`, `spirv-to-ir-representation`, `spirv-in-ir-representation`?

Reply:
`spirv-target-builtins` with description talking about how to represent _SPIR-V instructions_, rather than _SPIR-V Builtins_ looks good to me as well

Reply:
> I agree here. What about even more generic spirv-ops-target

Building on top of that suggestion: `spirv-target-env`?  It's nice and short, and implies that the output will be targeted towards a particular environment.

Reply:
> Building on top of that suggestion: spirv-target-env? It's nice and short, and implies that the output will be targeted towards a particular environment.

Looks nice!

We might even consider adding additional validation based on this option, using [OpenCL SPIR-V Environment spec](https://www.khronos.org/registry/OpenCL/specs/2.2/html/OpenCL_Env.html) as reference

Reply:
+1